jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunMojo.java

```diff
@@ -308,15 +315,86 @@ public void configureWebApplication() throws Exception
                 }
             }
         }
+        List<Overlay> warProjectConfiguration = getWarProjectConfiguration();
+        configureWebInf(warProjectConfiguration);
+        configureBaseResource(warProjectConfiguration);
 
-        //process any overlays and the war type artifacts
-        List<Overlay> overlays = getOverlays();
-        unpackOverlays(overlays); //this sets up the base resource collection
+        // this is to handle WARs that are not referenced in maven reactor
+        List<Overlay> overlaysToUnpack = warProjectConfiguration.stream()
+            .filter(e -> !e.isMavenProject())
+            .collect(Collectors.toList());
+        unpackOverlays(overlaysToUnpack);
 
         getLog().info("web.xml file = " + webApp.getDescriptor());
         getLog().info("Webapp directory = " + webAppSourceDirectory.getCanonicalPath());
     }
 
+    private void configureWebInf(List<Overlay> warProjectConfiguration)
+        throws IOException
+    {
+        Set<File> webInfLib = new LinkedHashSet<>();
+        Set<File> webInfClasses = new LinkedHashSet<>();
+        for (Overlay overlay : warProjectConfiguration)
+        {
+            Artifact artifact = overlay.getArtifact();
+            MavenProject project = mavenProjectHelper.getMavenProject(artifact);
+            if (project != null)
+            {
+                getLog().debug("WEB-INF/lib of " + artifact.getArtifactId() + " resolved from maven reactor");
+                getWebInfLibArtifacts(project).stream()
+                    .map(mavenProjectHelper::getArtifactPath)
+                    .map(Path::toFile)
+                    .forEach(webInfLib::add);
+                webInfClasses.add(mavenProjectHelper.getArtifactPath(artifact).toFile());
+            }
+            else
+            {
+                getLog().debug("WEB-INF/lib of " + artifact.getArtifactId() + " will be unpacked to target jetty_overlays");
+            }
+        }
+        webInfClasses.remove(webApp.getClasses());
+        webInfClasses.remove(webApp.getTestClasses());
+
+        webApp.getWebInfLib().addAll(webInfLib);
+        webApp.getWebInfClasses().addAll(webInfClasses);
```

Comment:
Even though the dependencies of the overlaid war are in the reactor as classes, they should still be put into the WEB-INF/lib of the webapp, not into the WEB-INF/classes.  See what we do already for the dependencies of the webapp that are jars.

Reply:
Alright 👍